pipeline_steps/common.groovy

```diff
@@ -301,7 +301,18 @@ String container_name(){
 def archive_artifacts(Map args = [:]){
   stage('Compress and Publish Artifacts'){
     try{
-      artifacts_dir = args.get("artifacts_dir", "${env.WORKSPACE}/artifacts")
+      // If the build fails, or this is a PR, the default
+      // artifact types to upload are 'log'. Otherwise we
+      // will attempt all types.
+      if ((currentBuild.result == "FAILURE") || (env.'ghprbPullId') != '') {
```

Comment:
I think you should actually be checking that `env.'ghprbPullId') != null` ... sorry, didn't catch that the first time around!